src/main/java/com/artipie/SliceFromConfig.java

```diff
@@ -0,0 +1,139 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.artipie;
+
+import com.artipie.composer.http.PhpComposer;
+import com.artipie.files.FilesSlice;
+import com.artipie.gem.GemSlice;
+import com.artipie.http.Response;
+import com.artipie.http.Slice;
+import com.artipie.maven.http.MavenSlice;
+import com.artipie.npm.Npm;
+import com.artipie.npm.http.NpmSlice;
+import com.artipie.rpm.http.RpmSlice;
+import com.jcabi.log.Logger;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+import org.cactoos.map.MapEntry;
+import org.cactoos.map.MapOf;
+import org.reactivestreams.Publisher;
+
+/**
+ * Slice from repo config.
+ * @since 0.1.4
+ * @todo #90:30min We still don't have tests for Pie. But now that this class was extracted, we have
+ *  a more cohesive class that could be tested. Write unit tests for SliceFromConfig class.
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+public final class SliceFromConfig implements Slice {
+
+    /**
+     * Repository config.
+     */
+    private final RepoConfig config;
+
+    /**
+     * Ctor.
+     * @param config Repo config
+     */
+    public SliceFromConfig(final RepoConfig config) {
+        this.config = config;
+    }
+
+    @Override
+    public Response response(final String line, final Iterable<Map.Entry<String, String>> headers,
+        final Publisher<ByteBuffer> body) {
+        try {
+            return SliceFromConfig.build(this.config).response(
+                line, headers, body
+            );
+        } catch (final InterruptedException ex) {
+            Logger.error(this, "Interruption when getting slice from config");
+            throw new IllegalArgumentException(ex);
+        } catch (final ExecutionException ex) {
+            Logger.error(this, "Exception when getting slice from config");
+            throw new IllegalArgumentException(ex);
+        }
+    }
+
+    /**
+     * Find a slice implementation for config.
+     * @param cfg Repository config
+     * @return Slice
+     * @throws ExecutionException If error getting the slice
+     * @throws InterruptedException If error getting the slice
+     * @todo #90:30min This method still needs more refactoring. First, we should extract
+     *  PhpComposer Slice construction to a private method to make it more readable. And then,
+     *  we should test if the type exist in the constructed map. If the type does not exist,
+     *  we should throw an IllegalStateException with the message "Unsupported repository type '%s'"
+     */
+    private static Slice build(final RepoConfig cfg) throws InterruptedException,
```

Comment:
@HDouss why do we have `InterruptedException` here? Does this method perform any blocking operation? We can't allow thread blocking for `Slice` implementations.